File: blockchain.go
package main

import (
	"bytes"
	"crypto/sha256"
	"fmt"
	"strconv"
	"time"
)

type block struct {
	index     int
	timestamp int64
	data      []byte
	prevHash  []byte
	hash      []byte
}

func (b *block) calculateHash() {
	timestamp := []byte(strconv.FormatInt(b.timestamp, 10))
	headers := bytes.Join([][]byte{b.prevHash, b.data, timestamp}, []byte{})
	hash := sha256.Sum256(headers)
	b.hash = hash[:]
}

type blockchain struct {
	chain []*block
}

func (bc *blockchain) init() {
	newBlock := block{index: 0, timestamp: time.Now().Unix(), prevHash: []byte(""), data: []byte("New chain")}
	newBlock.calculateHash()
	bc.chain = append(bc.chain, &newBlock)
}

func (bc *blockchain) addBlock(newBlock *block) {
	newBlock.prevHash = bc.chain[len(bc.chain)-1].hash
	newBlock.calculateHash()
	bc.chain = append(bc.chain, newBlock)
}

func main() {
	coin := blockchain{}
	coin.init()
	for i := 1; i <= 10; i++ {
		coin.addBlock(&block{index: 1, timestamp: time.Now().Unix(), data: []byte(fmt.Sprintf("amount %d", 8*i))})
	}

	for _, val := range coin.chain {
		fmt.Println("prevHash: " + string(val.prevHash))
		fmt.Println("Index: " + string(val.index))
		fmt.Println("Data: " + string(val.data))
		fmt.Println("Timestamp: " + string(val.timestamp))
		fmt.Println("hash: " + string(val.hash))
		fmt.Println("--------------------------------------------------------")
	}

}

File: blockchain.js
const SHA256 = require("crypto-js/sha256");

const Block = function(index, data, timestamp = new Date(), prevHash = ""){

    Block.prototype.calculateHash = function(){
        return SHA256(this.index + this.timestamp + this.prevHash + JSON.stringify(this.data)).toString();
    };
    this.index = index;
    this.data = data;
    this.timestamp = timestamp;
    this.prevHash = prevHash;
    this.hash = this.calculateHash();
};


const Blockchain = function(){

    Blockchain.prototype.createGensisBLock = function(){
        return new Block(0, "Genesis block"); 
    };

    Blockchain.prototype.getLatestBlock = function(){
        return this.chain[this.chain.length - 1];
    };

    Blockchain.prototype.addBlock = function(newBlock) { 
        newBlock.prevHash = this.getLatestBlock().hash; 
        newBlock.hash = newBlock.calculateHash(); 
        this.chain.push(newBlock); 
    };

    Blockchain.prototype.isChainValid = function() { 
        for (let i = 1; i < this.chain.length; i++){ 
            const currentBlock = this.chain[i]; 
            const prevBlock = this.chain[i - 1]; 
            if (currentBlock.hash !== currentBlock.calculateHash()) { 
                return false; 
            } 

            if (currentBlock.prevHash !== prevBlock.hash) { 
                return false; 
            } 
        } 
        return true; 
    } 
    this.chain = [this.createGensisBLock()];
}

const coin = new Blockchain()
for(let i = 1; i <= 10; i++){
    coin.addBlock(new Block(i, {amount: 8 * i}));
}
console.log(coin.chain);